core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -406,60 +400,61 @@ private void deleteExistingUserToOverwriteIt(String mobileNumber, String userPro
 		// because the anonymized data cannot be retrieved
 		// (the relation is encrypted, the password is not available)
 		userRepository.delete(existingUserEntity);
+		userRepository.flush(); // So we can insert another one
 		logger.info("User with mobile number '{}' and ID '{}' removed, to overwrite the account",
 				existingUserEntity.getMobileNumber(), existingUserEntity.getId());
 	}
 
-	@Transactional
+	@Transactional(dontRollbackOn = MobileNumberConfirmationException.class)
 	public UserDto confirmMobileNumber(UUID userId, String userProvidedConfirmationCode)
 	{
-		User userEntity = getUserEntityById(userId);
-		ConfirmationCode confirmationCode = userEntity.getMobileNumberConfirmationCode();
+		User updatedUserEntity = updateUser(userId, userEntity -> {
+			ConfirmationCode confirmationCode = userEntity.getMobileNumberConfirmationCode();
 
-		assertValidConfirmationCode(userEntity, confirmationCode, userProvidedConfirmationCode,
-				() -> MobileNumberConfirmationException.confirmationCodeNotSet(userEntity.getMobileNumber()),
-				r -> MobileNumberConfirmationException.confirmationCodeMismatch(userEntity.getMobileNumber(),
-						userProvidedConfirmationCode, r),
-				() -> MobileNumberConfirmationException.tooManyAttempts(userEntity.getMobileNumber()));
+			assertValidConfirmationCode(userEntity, confirmationCode, userProvidedConfirmationCode,
```

Comment:
This method name could be made more speaking to reflect that it does an update underneath.

Reply:
I'm not disagreeing, but I doubt whether the code becomes more readable if we would change this into something like ``assertValidConfirmationCodeAndRegisterAttempt``.